src/Profile/PageProfile.cpp

```diff
@@ -40,47 +40,47 @@ static void
 Load(const ProfileMap &map, PageLayout &_pl, const unsigned page)
 {
   char profileKey[32];
-  unsigned prefixLen = sprintf(profileKey, "Page%u", page);
-  if (prefixLen <= 0)
-    return;
-
-  PageLayout pl = PageLayout::Default();
-  strcpy(profileKey + prefixLen, "InfoBoxMode");
-  if (!map.Get(profileKey, pl.infobox_config.auto_switch))
-    return;
-  strcpy(profileKey + prefixLen, "InfoBoxPanel");
-  if (!map.Get(profileKey, pl.infobox_config.panel))
-    return;
-
-  strcpy(profileKey + prefixLen, "Layout");
-  unsigned temp = 0;
-  map.Get(profileKey, temp);
-  switch (temp) {
-  case tlEmpty:
-    pl.valid = false;
-    break;
-
-  case tlMap:
-    pl.infobox_config.enabled = false;
-    break;
+  int prefixLen = snprintf(profileKey, 32, "Page%u", page);
+  if (prefixLen > 0 && prefixLen <= 32)
+  {
+    PageLayout pl = PageLayout::Default();
+    strcpy(profileKey + prefixLen, "InfoBoxMode");
```

Comment:
Since you felt it was important to check the `prefixLen` for overflowing the buffer - what happens in this line if `prefixLen==31`? The `strcpy()` would overflow the buffer.
I don't feel like this overflow check has a point, but in the context of your decision that it does have a point, the rest of this commit doesn't make sense.